Compile grpcio_tools against system ABSL and protobuf instead
of statically linking another copy.

Improves our security posture by reducing duplicate code.
--- a/tools/distrib/python/grpcio_tools/setup.py
+++ b/tools/distrib/python/grpcio_tools/setup.py
@@ -43,6 +43,9 @@
 
 _EXT_INIT_SYMBOL = "PyInit__protoc_compiler"
 
+BUILD_WITH_SYSTEM_ABSL = True
+BUILD_WITH_SYSTEM_PROTOBUF = True
+
 _parallel_compile_patch.monkeypatch_compile_maybe()
 _spawn_patch.monkeypatch_spawn()
 
@@ -260,6 +263,16 @@
 GRPC_PYTHON_TOOLS_PACKAGE = "grpc_tools"
 GRPC_PYTHON_PROTO_RESOURCES_NAME = "_proto"
 
+if BUILD_WITH_SYSTEM_ABSL:
+    CC_FILES = [x for x in CC_FILES if "third_party/abseil-cpp" not in x]
+    CC_INCLUDES = [x for x in CC_INCLUDES if "third_party/abseil-cpp" not in x]
+
+if BUILD_WITH_SYSTEM_PROTOBUF:
+    CC_FILES = [x for x in CC_FILES if "third_party/protobuf" not in x]
+    CC_INCLUDES = [x for x in CC_INCLUDES if "third_party/protobuf" not in x]
+    EXTRA_LINK_ARGS.extend(["-lprotoc", "-lprotobuf"])
+    PROTO_INCLUDE = "/usr/include"
+
 DEFINE_MACROS = ()
 if "win32" in sys.platform:
     DEFINE_MACROS += (
